# Publish all joint states
joint_state_controller:
  type: "joint_state_controller/JointStateController"
  publish_rate: 250

arm_controller:
  type: "position_controllers/JointTrajectoryController"
  publish_rate: 250
  joints:
    - crane_x7_shoulder_fixed_part_pan_joint
    - crane_x7_shoulder_revolute_part_tilt_joint
    - crane_x7_upper_arm_revolute_part_twist_joint
    - crane_x7_upper_arm_revolute_part_rotate_joint
    - crane_x7_lower_arm_fixed_part_joint
    - crane_x7_lower_arm_revolute_part_joint
    - crane_x7_wrist_joint
  constraints:
    goal_time: 0.0
    stopped_velocity_tolerance: 1.0

gripper_controller:
  type: "position_controllers/GripperActionController"
  publish_rate: 250
  joint: crane_x7_gripper_finger_a_joint
  action_monitor_rate: 10
  state_publish_rate:  100
  stall_velocity_threshold: 0.01
  goal_tolerance: 0.2
  stall_timeout: 0.3

dynamixel_port:
  port_name: "/dev/ttyUSB0"
  baud_rate: 3000000
  joints:
    - crane_x7_shoulder_fixed_part_pan_joint
    - crane_x7_shoulder_revolute_part_tilt_joint
    - crane_x7_upper_arm_revolute_part_twist_joint
    - crane_x7_upper_arm_revolute_part_rotate_joint
    - crane_x7_lower_arm_fixed_part_joint
    - crane_x7_lower_arm_revolute_part_joint
    - crane_x7_wrist_joint
    - crane_x7_gripper_finger_a_joint
  crane_x7_shoulder_fixed_part_pan_joint: {id: 2, center: 2048, home: 2048, effort_const: 1.79, mode: 3 }
  crane_x7_shoulder_revolute_part_tilt_joint: {id: 3, center: 2048, home: 2048, effort_const: 2.79, mode: 3 }
  crane_x7_upper_arm_revolute_part_twist_joint: {id: 4, center: 2048, home: 2048, effort_const: 1.79, mode: 3 }
  crane_x7_upper_arm_revolute_part_rotate_joint: {id: 5, center: 2048, home: 2048, effort_const: 1.79, mode: 3 }
  crane_x7_lower_arm_fixed_part_joint: {id: 6, center: 2048, home: 2048, effort_const: 1.79, mode: 3 }
  crane_x7_lower_arm_revolute_part_joint: {id: 7, center: 2048, home: 2048, effort_const: 1.79, mode: 3 }
  crane_x7_wrist_joint: {id: 8, center: 2048, home: 2048, effort_const: 1.79, mode: 3 }
  crane_x7_gripper_finger_a_joint: {id: 9, center: 2048, home: 2048, effort_const: 1.79, mode: 3 }

joint_limits:
  arm_joint1:
    has_position_limits: true
    min_position: -1.5707963268
    max_position: 1.5707963268
    has_velocity_limits: true
    max_velocity: 5.969211435
    has_acceleration_limits: false
    has_jerk_limits: false
    has_effort_limits: true
    max_effort: 4.0
  arm_joint2:
    has_position_limits: true
    min_position: -1.5707963268
    max_position: 0.0
    has_velocity_limits: true
    max_velocity: 5.969211435
    has_acceleration_limits: false
    has_jerk_limits: false
    has_effort_limits: true
    max_effort: 4.0
  arm_joint3:
    has_position_limits: true
    min_position: -1.5707963268
    max_position: 1.5707963268
    has_velocity_limits: true
    max_velocity: 5.969211435
    has_acceleration_limits: false
    has_jerk_limits: false
    has_effort_limits: true
    max_effort: 4.0
  arm_joint4:
    has_position_limits: true
    min_position: 0.0
    max_position: 2.726204
    has_velocity_limits: true
    max_velocity: 5.969211435
    has_acceleration_limits: false
    has_jerk_limits: false
    has_effort_limits: true
    max_effort: 4.0
  arm_joint5:
    has_position_limits: true
    min_position: -1.5707963268
    max_position: 1.5707963268
    has_velocity_limits: true
    max_velocity: 5.969211435
    has_acceleration_limits: false
    has_jerk_limits: false
    has_effort_limits: true
    max_effort: 4.0
  arm_joint6:
    has_position_limits: true
    min_position: -2.094395
    max_position: 1.047196
    has_velocity_limits: true
    max_velocity: 5.969211435
    has_acceleration_limits: false
    has_jerk_limits: false
    has_effort_limits: true
    max_effort: 4.0
  arm_joint7:
    has_position_limits: true
    min_position: -2.967060
    max_position: 2.967060
    has_velocity_limits: true
    max_velocity: 5.969211435
    has_acceleration_limits: false
    has_jerk_limits: false
    has_effort_limits: true
    max_effort: 4.0
  hand_joint1:
    has_position_limits: true
    min_position: 0.0
    max_position: 0.523598776
    has_velocity_limits: true
    max_velocity: 5.969211435
    has_acceleration_limits: false
    has_jerk_limits: false
    has_effort_limits: true
    max_effort: 4.0
